src/lib/parser.rs

```diff
@@ -108,8 +108,47 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
-    let letter = self.contents.get(self.index)?;
+    let letter = *self.contents.get(self.index)? as char;
+
+    let fs = '/' as u8;
+    let nl = '\n' as u8;
+    let ast = '*' as u8;
```

Comment:
You don't need these anymore now. 
in staid of:
```rust
     if letter == &fs { 
```
You can just do:
```rust
     if letter == '/' { 
```
This does not require variable assignments, requires less code and makes it more readable